### ISSUE TITLE
[docs][byob] docs for tool writers

### ISSUE THREAD

**Issue by asraa** [slsa-framework/slsa-github-generator#1574]
**Is your feature request related to a problem? Please describe.**
Start docs aimed towards tool writers to create their own SLSA 3 builder.

Some notable items and example we should provide:
* An example of a composite tool action of an existing action hosted in their repo (this should be referenced as a local action in order to ensure that the action is called at the correct reference)
* Passing write permissions through a `github.token` for various permissions: packages, contents
* Passing secret(s)
* An example tool that produces multiple attestations
* An example TRW that publishes outside of the tool action
* An Example PW which calls in parallel multiple TRW
* The output builder ID referencing their tool reusable workflow
* Verification with slsa-verifier
* Performing authentication (e.g. docker registry, GCP) inside a tool action
* TRW Action outputs of different types: array, json, string
* Private project repositories

All of these will need corresponding e2e tests, maybe under `example-byob` or some other name?

cc @laurentsimon

**Comment by laurentsimon** [comment]
- secrets don't support defaults in reusable workflows. Need to show how to use use defaults `github.token`
- inputs need a type

**Comment by laurentsimon** [comment]
note: I'm doing a Syft PoC at https://github.com/laurentsimon/sbom-action/blob/feat/slsa-trw/.github/workflows/slsa3.yml

**Comment by ianlewis** [comment]
Related to #1552 

I started a template example repo at https://github.com/ianlewis/slsa-byob-template

@asraa @laurentsimon We can maybe move the repo under the SLSA org and start iterating on docs there. wdut?

**Comment by ianlewis** [comment]
With all the docs we might have to write, maybe we should start thinking about setting up a proper website with rendered HTML docs...

**Comment by laurentsimon** [comment]
Do you think GitHub pages would be a good start? 
Do you think we could host the doc in this repo under a different branch with no branch protection enabled? Or ding it on another repo is simpler?

Shall we brainstorm the high-level sections we have in mind before?
0. Intro
 - Terminology
 - Supported triggers
2. Getting started
- I want to wrap an existing Action
- I want to create a compliant builder from scratch
3. Maintenance
 - Use X
 - Use Y
2. Improve my security
 - Generate provenance for my binaries
 - Verify them in Action
 - Follow Scorecard best practice
4. Let your users know
 - Update your README
 - Opt in the compliance program
5. FAQs / Common pitfalls
 - Error msg "bla"
 - 
6. Examples
 - Passing secrets
 - Using GH token

I'm just making this up :)

**Comment by ianlewis** [comment]
I think gh pages would be great. We don't have a lot of collective bandwidth and expertise for docs so I think we'll want to use something with low friction.

The slsa website uses netlify so that might be another option to look at.

**Comment by laurentsimon** [comment]
OK. I propose starting with .md files. Hopefully GH pages will allow us to turn it into a website with a pre-defined theme with little effort if we want to polish it.

**Comment by ianlewis** [comment]
> OK. I propose starting with .md files. Hopefully GH pages will allow us to turn it into a website with a pre-defined theme with little effort if we want to polish it. Wdut?

Yeah, it should be easy to do. Definitely something we have to lead with. We can defer decisions until we need to polish and organize. the docs

**Comment by ianlewis** [comment]
Related to #892

**Comment by laurentsimon** [comment]
Closing this as completed, see https://github.com/slsa-framework/slsa-github-generator/blob/main/BYOB.md. We will iterate and update the README to mention BYOB in follow-up PR
